### PR TITLE
[16.0][IMP] edi_storage_oca: new opt-in technical field to avoid moving files after processing

### PR DESCRIPTION
**Pull request by thienvh332**

### CODE REVIEW

File: edi_storage_oca/components/listener.py [hunk 1]
```diff
@@ -55,12 +55,13 @@ def on_edi_exchange_done(self, record):
             ).as_posix()
             if not done_dir:
                 return res
-            res = self._move_file(storage, pending_dir, done_dir, file)
-            if not res:
-                # If a file previously failed it should have been previously
-                # moved to the error dir, therefore it is not present in the
-                # pending dir and we need to retry from error dir.
-                res = self._move_file(storage, error_dir, done_dir, file)
+            if record.move_file_after_processing:
```
Comment (nilshamerlinck) [inline edi_storage_oca/components/listener.py]: why not in the first condition? (above)
`if record.direction == "input" and storage and record.move_file_after_processing`

File: edi_storage_oca/components/listener.py [hunk 2]
```diff
@@ -74,6 +75,6 @@ def on_edi_exchange_error(self, record):
             error_dir = record.type_id._storage_fullpath(
                 record.backend_id.input_dir_error
             ).as_posix()
-            if error_dir:
+            if error_dir and record.move_file_after_processing:
```
Comment (nilshamerlinck) [inline edi_storage_oca/components/listener.py]: same, why not in the first condition?

Comment (thienvh332) [reply edi_storage_oca/components/listener.py]: I will update it

**Comment by nilshamerlinck** [comment]
`[16.0][UPD] edi_storage_oca: Do not move files to storage with the endpoint`
->
`[16.0][IMP] edi_storage_oca: new opt-in technical field to avoid moving files after processing`

File: edi_storage_oca/components/listener.py [hunk 3]
```diff
@@ -40,6 +40,8 @@ def _add_post_commit_hook(
         )
 
     def on_edi_exchange_done(self, record):
+        if not record.move_file_after_processing:
```
Comment (nilshamerlinck) [inline edi_storage_oca/components/listener.py]: why so early? it doesn't make sense
there is a logical hierarchy in the check:
- 1) check record.direction
- 2) then check if a storage is defined
- 3) and then check if we move_file_after_processing

**Review by nilshamerlinck — CHANGES_REQUESTED** [review]

File: edi_storage_oca/components/listener.py (hunk 3 above)
Comment (thienvh332) [reply edi_storage_oca/components/listener.py]: What I see here is the "And" condition, which means checking what comes before and after is the same. But I fixed it adding your suggestion. Thanks nils

Comment (nilshamerlinck) [reply edi_storage_oca/components/listener.py]: it is the same but sometimes code readability matters, thanks

**Review by nilshamerlinck — APPROVED** [review]

**Comment by nilshamerlinck** [comment]
in the OCA PR description, please explain things a bit otherwise they won't get it:

In some contexts, we might have an EDI Exchange Type with both:
- a storage (e.g. files pushed to a SFTP)
- AND an endpoint

If a record is created through the endpoint, we want to disable the logic that moves files after processing (implemented as a listener:
https://github.com/oca/edi-framework/blob/16.0/edi_storage_oca/components/listener.py#L13)

**Pull request merged**